examples/webgl_materials_texture3d_volume1.html

```diff
@@ -0,0 +1,308 @@
+				});
+
+			} );
+
+			// renderer
+			var canvas = document.createElement( 'canvas' );
+			var context = canvas.getContext( 'webgl2' );
+			renderer = new THREE.WebGLRenderer( { canvas: canvas, context: context } );
+			renderer.setPixelRatio( window.devicePixelRatio );
+			renderer.setSize( window.innerWidth, window.innerHeight );
+
+			container = document.createElement( 'div' );
+			document.body.appendChild( container );
+			container.appendChild( renderer.domElement );
+
+			controls = new THREE.OrthographicTrackballControls( camera, renderer.domElement );
+			controls.target.set( 64, 64, 128 );
+			controls.rotateSpeed = 5.0;
+			controls.zoomSpeed = 5;
+			controls.panSpeed = 2;
+			controls.noZoom = false;
+			controls.noPan = false;
+			controls.staticMoving = true;
+			controls.dynamicDampingFactor = 0.3;
+
+			stats = new Stats();
+			container.appendChild( stats.dom );
+
+			var gui = new dat.GUI();
+
+			setupInset();
+
+			window.addEventListener( 'resize', onWindowResize, false );
+
+		}
+
+		function updateUniforms() {
+
+			config = this.volumeConfig;
+
+			this.volmaterial.uniforms.u_clim = {type: 'v2', value: [config.clim1, config.clim2]};
+			this.volmaterial.uniforms.u_renderstyle = {type: 'int', value: config.renderstyle == 'mip' ? 0 : 1}; // 0: MIP, 1: ISO
+			this.volmaterial.uniforms.u_renderthreshold = {type: 'f', value: config.isothreshold};  // For ISO renderstyle
+			this.volmaterial.uniforms.u_cmdata = { type: 't', value: this.cmtextures[config.colormap] };
+
+			this.volmaterial.needsUpdate = true;
+
+		}
+
+		function onWindowResize() {
+
+			this.camera.right = window.innerWidth / 5.0;
+			this.camera.left = -this.camera.right;
+			this.camera.top = window.innerHeight / 5.0;
+			this.camera.bottom = -this.camera.top;
+			camera.updateProjectionMatrix();
+
+			renderer.setSize( window.innerWidth, window.innerHeight );
+
+			controls.handleResize();
+
+		}
+
+		function animate() {
+
+			requestAnimationFrame( animate );
+
+			controls.update();
+
+			//copy position of the camera into inset
+			camera2.position.copy( camera.position );
+			camera2.position.sub( controls.target );
+			camera2.position.setLength( 300 );
+			camera2.lookAt( scene2.position );
+
+			renderer.render( scene, camera );
+			renderer2.render( scene2, camera2);
+
+			stats.update();
+
+		}
+
+		function rotateAroundWorldAxis(object, axis, radians) {
+			var rotWorldMatrix = new THREE.Matrix4();
+			rotWorldMatrix.makeRotationAxis(axis.normalize(), radians);
+
+			object.applyMatrix(rotWorldMatrix);
+
+		}
+
+		function setupInset () {
+			var insetWidth = 150,
+				insetHeight = 150;
+			container2 = document.getElementById('inset');
+			container2.width = insetWidth;
+			container2.height = insetHeight;
+
+			// renderer
+			renderer2 = new THREE.WebGLRenderer( {alpha : true} );
+			renderer2.setClearColor( 0x000000, 0 );
+			renderer2.setSize( insetWidth, insetHeight );
+			container2.appendChild( renderer2.domElement );
```

Comment:
I haven't read the code in the detail yet but what's `renderer2` for?

Reply:
Basically, I toke another example to have a starting point for this demo (I'm quite new to Three, as you might have guessed :)). I guess this is one of the oddities that came with it. All the xx2 stuff is to draw a small coordinate frame in the bottom right. I removed it, making the code less complex.